services/server/src/server/services/workers/verificationWorker.ts

```diff
@@ -63,16 +67,41 @@ const initWorker = () => {
   vyper = new VyperLocal(Piscina.workerData.vyperRepoPath);
 };
 
-export async function verifyFromJsonInput({
+async function runWorkerFunction<T extends VerificationWorkerInput>(
+  workerFunction: (input: T) => Promise<VerifyOutput>,
+  input: T,
+): Promise<VerifyOutput> {
+  initWorker();
+  const context = { traceId: input.traceId };
+  return asyncLocalStorage.run(context, workerFunction, input);
```

Comment:
I see this now runs the worker funciton with this context. But I couldn't see where 
1. This context is being used 
2. Where the traceId is being logged.

Doesn't the `traceId` need to be within these fields? Maybe TS is missing this because it's run with `asynclocalstorage.run()`?
https://github.com/ethereum/sourcify/blob/8eba637e823cd955ff04f9f7bb6c9fad822f7310/services/server/src/server/services/workers/verificationWorker.ts#L97-L105

From what I can see the traceId is passed at VerificationService in the `input` when calling each worker but I don't see it anywhere after that.

Reply:
Our logger module gets the trace automatically:

https://github.com/ethereum/sourcify/blob/8eba637e823cd955ff04f9f7bb6c9fad822f7310/services/server/src/common/logger.ts#L71-L75

Since the worker is running in it's own thread, we need to set it inside the worker again (server sets it per request for the main thread).

The logging happens here:

https://github.com/ethereum/sourcify/blob/8eba637e823cd955ff04f9f7bb6c9fad822f7310/services/server/src/common/logger.ts#L100-L110